### PR TITLE
mesa: update to 19.3.1.

### PR DESCRIPTION
**Pull request by travankor**
Compiles:
- [x]  x86_64

- [x]  x86_64-musl

- [x]  i386

- [x] aarch64

- [x] aarch64-musl

- [x]  armv7

- [x] armv7-musl

Tested:

- [x] x86_64: Intel

- [x] x86_64-musl: Nouveau

### CODE REVIEW

**Comment by q66** [comment]
19.3.1 is already out, we should update to that instead

File: srcpkgs/mesa/patches/add-use-elf-tls.patch [hunk 1]
```diff
@@ -1,28 +0,0 @@
...
-index 4afd8ca..ab9b365 100644
---- meson.build
-+++ meson.build
-@@ -378,7 +378,8 @@ if with_egl and not (with_platform_drm or with_platform_surfaceless or with_plat
- endif
- 
- # Android uses emutls for versions <= P/28. For USE_ELF_TLS we need ELF TLS.
--if not with_platform_android or get_option('platform-sdk-version') >= 29
-+with_use_elf_tls = get_option('use-elf-tls')
-+if with_use_elf_tls and not with_platform_android or get_option('platform-sdk-version') >= 29
```
Comment (st3r4g) [inline srcpkgs/mesa/patches/add-use-elf-tls.patch]: Please don't remove this patch, musl will break (issue https://gitlab.freedesktop.org/mesa/mesa/issues/966). This line needs to be adapted to include the changes (they added windows to the non-USE_ELF_TLS platforms).

**Comment by travankor** [comment]
Thanks for the info!
Though, I'm running into this error when cross-building to `aarch64-musl`:
```

```

**Comment by q66** [comment]
Well, zap your masterdir and try again I guess...

Also, LLVM 9.0.1 is out, we should update it before this. That should resolve stability issues on AMD Navi.

**Comment by travankor** [comment]
>Well, zap your masterdir and try again I guess...

Already tried that. I think I have to create a new branch so that xbps-src reinstalls things

>Also, LLVM 9.0.1 is out, we should update it before this. That should resolve stability issues on AMD Navi.

Sounds good to me.

**Comment by pullmoll** [comment]
The `MesaLib-devel` subpackage is missing a dependency on `mesa>=${version}_${revision}` or `${sourcepkg}>=${version}_${revision}`.
If you look into `xbps-query -Rf MesaLib-devel` there is a symlink from `libGLX_mesa.so -> libGLX_mesa.so.0` and that target is contained only in `mesa-${version}`.

**Comment by q66** [comment]
LLVM 9.0.1 tracking here https://github.com/void-linux/void-packages/pull/17671

**Comment by travankor** [comment]
I'm not sure if I should split this into a separate PR, but I noticed that most of the FOSS ARM GPU drivers are not built by default. I've added a commit that enables these along with a few examples of what hardware uses the drivers.

**Comment by xtraeme** [comment]
Will merge after the python 3.8.1 rebuild.

**Pull request merged**